storm/check.py

```diff
@@ -0,0 +1,688 @@
+                        for ks in _g(s, [], None, sc):
+                            stream_id = _g(ks, 'unknown', None, 'stream_id')
+                            ks_tags = k_tags + ['stream:{}'.format(stream_id)]
+                            component_id = _g(ks, None, None, 'component_id')
+                            if component_id:
+                                ks_tags.append('component:{}'.format(component_id))
+
+                            component_value = _g(ks, 0.0, _float, 'value')
+                            if component_value is not None:
+                                # will make stats like these two examples
+                                # storm.topologyStats.metrics.spouts.emitted
+                                # storm.topologyStatus.metrics.bolts.acked
+                                k_stat['storm.topologyStats.metrics.{}.{}'.format(k, sc)].append(
+                                    (component_value, ks_tags))
+                    r[k].append(k_stat)
+            return r
+
+    def report_gauge(self, metric, value, tags, additional_tags=list()):
+        """ Report the Gauge Metric.
+
+        :param metric:
+        :param value:
+        :param tags:
+        :param additional_tags:
+        :return:
+        """
+        all_tags = tags + [
+            'env:{}'.format(self.environment_name),
+            'environment:{}'.format(self.environment_name)] + additional_tags
+        self.gauge(
+            metric=metric,
+            value=value,
+            tags=all_tags
+        )
+
+    def report_histogram(self, metric, value, tags, additional_tags=list()):
+        """ Report the Histogram Metric.
+
+        :param metric:
+        :param value:
+        :param tags:
+        :param additional_tags:
+        :return:
+        """
+        all_tags = tags + [
+            'env:{}'.format(self.environment_name),
+            'environment:{}'.format(self.environment_name)] + additional_tags
+        self.histogram(
+            metric=metric,
+            value=value,
+            tags=all_tags
+        )
+
+    def update_from_config(self, instance):
+        """ Update Configuration tunables from instance configuration.
+
+        :param instance: Agent config instance.
+        :return: None
+        """
+        if instance.get('https', self.init_config.get('https', "False")).lower() in ['true', 't', '1']:
+            self.http_prefix = 'https://'
+        else:
+            self.http_prefix = 'http://'
+        self.nimbus_server = instance.get('server', self.init_config.get('server', 'localhost:9005'))
+        self.environment_name = instance.get('environment', self.init_config.get('environment', 'dev'))
+        self.additional_tags = instance.get('tags', [])
+        self.excluded_topologies = instance.get('excluded', [])
+        intervals = instance.get('intervals', self.init_config.get('intervals', [60]))
+        self.intervals = []
+        if isinstance(intervals, (list, tuple)):
```

Comment:
You don't need to introspect `intervals` type and convert them. We can state in the configuration we're expecting a list of int and raise an error if it's not the case.